assets/js/amp-customize-controls.js

```diff
@@ -0,0 +1,164 @@
+	api.state.add( 'ampAvailable', new api.Value() ).set( false );
+
+	/**
+	 * Check if the URL is AMPified.
+	 *
+	 * @param {string} url URL.
+	 * @return {boolean} whether it is an AMP URL.
+	 */
+	function isAmpUrl( url ) {
+		var urlParser = document.createElement( 'a' ),
+			regexParam = new RegExp( "(^|\\?|&)" + ampVars.query + "=1(?=&|$)" ),
+			regexEndpoint = new RegExp( "\\/" + ampVars.query + "\\/?$" );
+
+		urlParser.href = url;
+		if ( regexParam.test( urlParser.search ) ) {
+			return true;
+		}
+		return regexEndpoint.test( urlParser.pathname );
+	}
+
+	/**
+	 * Create an non-AMP version of a URL.
+	 *
+	 * @param {string} url URL.
+	 * @return {string} non-AMPified URL.
+	 */
+	function unampifyUrl( url ) {
+		var urlParser = document.createElement( 'a' ),
+			regexParam = new RegExp( "(^|\\?|&)" + ampVars.query + "=1" ),
+			regexEndpoint = new RegExp( "\\/" + ampVars.query + "\\/?$" );
+
+		urlParser.href = url;
+		urlParser.pathname = urlParser.pathname.replace( regexEndpoint, '' );
+		urlParser.search = urlParser.search.replace( regexParam, '' );
+		return urlParser.href;
+	}
+
+	/**
+	 * Create an AMP version of a URL.
+	 *
+	 * @param {string} url URL.
+	 * @return {string} AMPified URL.
+	 */
+	function ampifyUrl( url ) {
+		var urlParser = document.createElement( 'a' );
+		urlParser.href = unampifyUrl( url );
+		if ( urlParser.search.length ) {
+			urlParser.search += '&';
+		}
+		urlParser.search += ampVars.query + '=1';
+		return urlParser.href;
+	}
+
+	/**
+	 * Hook up all AMP preview interactions once panel is ready.
+	 *
+	 * @param {wp.customize.Panel} panel The AMP panel.
+	 * @return {void}
+	 */
+	function panelReady( panel ) {
+		/**
+		 * Make current URL AMPified if toggle is on.
+		 *
+		 * @param {string} url URL.
+		 * @return {string} AMPified URL.
+		 */
+		function setCurrentAmpUrl( url ) {
+			var ampToggle = api.state( 'ampEnabled' ).get() && api.state( 'ampAvailable' ).get();
+			if ( ! ampToggle && isAmpUrl( url ) ) {
+				return unampifyUrl( url );
+			} else if ( ampToggle && ! isAmpUrl( url ) ) {
+				return ampifyUrl( url );
+			}
+			return url;
+		}
+
+		/**
+		 * Swap to AMP version of URL in preview.
+		 *
+		 * @return {void}
+		 */
+		function updatePreviewUrl() {
+			api.previewer.previewUrl.set( setCurrentAmpUrl( api.previewer.previewUrl.get() ) );
+		}
+
+		/**
+		 * Inspect iFrame metadata to see if loaded page was successfully AMP'd.
+		 *
+		 * @return {void}
+		 */
+		function isPageAmpified() {
+			// TODO, set true for now.
+			// Handle tooltip display in here if page isn't able to be AMP'd.
+			api.state( 'ampAvailable' ).set( true );
+		}
+
+		// AMP panel triggers the input toggle for AMP preview.
+		panel.expanded.bind( function() {
+			api.state( 'ampEnabled' ).set( panel.expanded.get() );
+		} );
+
+		// Open AMP panel if mobile device selected.
+		api.previewedDevice.bind( function( device ) {
+			panel.expanded.set( 'mobile' === device );
```

Comment:
Is this right? The AC says:

> When a user selects to view the mobile version of a page that is AMP supported, the new toggle will automatically default to AMP. The user will be able to change it to Original if they wish to do so.

I think this means that `ampEnabled` should be set to `true` when switching to the mobile `previewedDevice`. I would think it could be annoying for the expanded panel to change when the user just changes the previewed device, as it would remove the controls context for where they were previously working.